src/QsCompiler/CompilationManager/DiagnosticTools.cs

```diff
@@ -19,7 +19,7 @@ public static class DiagnosticTools
         /// Returns the line and character of the given position as tuple without verifying them.
         /// Throws an ArgumentNullException if the given position is null.
         /// </summary>
-        internal static Tuple<int, int> AsTuple(Position position) => 
+        public static Tuple<int, int> AsTuple(Position position) => 
```

Comment:
Perhaps a bigger change than this PR, but since this method enforces non-nullability of `position`, it may be helpful to add a `#nullable enable` line at the top of the file.

Reply:
Indeed we should make a bigger pass (and breaking change) to switch to using the nullable feature instead of some of our custom typing/checks. I'd rather address that then across the entire code base rather than just here.

Reply:
This case doesn't look like a breaking change since the `position` parameter doesn't use either `QsNullable` or `NonNullable`, so its type would stay the same.

`#nullable enable` seems to be made for cases like this, so that you can incrementally convert an existing codebase to the new nullable system without doing it all at once. We will need a full pass too but that pass will be easier if there are already files that have been converted earlier.

Reply:
As @samarsha notes, adding `#nullable enable` is by design non-breaking aside from the possible introduction of new warnings; that statement will not introduce new errors so as to allow for progressively adopting nullability checks.

Reply:
It still seems a bit orthogonal to this PR, since it is not related to any change. I think that's a good thing to do, but it would still make sense to make a pass on a larger scale.